### ISSUE TITLE
PCRE2 support still requires PCRE1

### ISSUE THREAD

**Issue by zhaoshikui** [SpiderLabs/ModSecurity#2966]
libmodsecurity 2.9.7 compiles fine using --with-pcre2 only when both pcre1 and pcre2 dev packages are installed, which is wrong...
Moreover, the resulting binary links against both libraries... which is further wrong.

ldd mod_security2.so |grep pcre
ldd: warning: you do not have execution permission for `./mod_security2.so'
	libpcre.so.1 => not found
	libpcre2-8.so.0 => /usr/lib64/libpcre2-8.so.0 (0x0000ffffb9194000)
        ...
If only pcre2 dev package is installed, it fails at ./configure --with-pcre2 saying:

checking for libpcre config script... no
configure: *** pcre library not found.
configure: error: pcre library is required

In other words, --with-pcre2 still requires pcre1 dev.
This question is the same as the [previous one](https://github.com/SpiderLabs/ModSecurity/issues/2750),however, this issue is not resolved in v2.9.7.

**Comment by martinhsv** [comment]
Hello @zhaoshikui ,

Just checking my config.log from when the relevant testing was done and I can see:
```
configure:16348: checking for libpcre config script
configure:16399: result: no
configure:16412: *** pcre library not found.
configure:16445: checking for libpcre2 config script
configure:16475: result: /usr/bin/pcre2-config
configure:16507: using pcre2 v10.39
```

... and a successful completion of the configure step.

Moreover, the ModSecurity v2 message 'pcre library is required' was removed here: https://github.com/SpiderLabs/ModSecurity/commit/8fc0b519b7a6c023259753a21f33bf3649a25b14

Could you please double-check what you've done? If you still see the same result, a more complete description of your build environment and steps to reproduce might be helpful.

**Comment by marcstern** [comment]
Compiling master on Rocky 9. pcre is still linked:
```
> ldd /usr/local/lib64/httpd/modules/mod_security2.so
        [...]
        libpcre.so.1 => /usr/lib64/libpcre.so.1 (0x00007f73ca1b4000)
        libpcre2-8.so.0 => /usr/lib64/libpcre2-8.so.0 (0x00007f73ca118000)
        [...]
```

**Comment by airween** [comment]
> Compiling master on Rocky 9. pcre is still linked:
> 
you mean the **current** mod_security2 [master](https://github.com/SpiderLabs/ModSecurity/commit/11f85b82e1e88d4b38157bb4d02d88110f31402e)?

With 2.9.7 on Debian 12 I get:
```
$ ldd /usr/lib/apache2/modules/mod_security2.so | grep pcre
	libpcre2-8.so.0 => /lib/x86_64-linux-gnu/libpcre2-8.so.0 (0x00007ff7be17b000)
```

only pcre2 dependency, after commit 05396552.

**Comment by marcstern** [comment]
Yes, latest v2/master.
Here are my configure flages: --enable-pcre-study --enable-pcre-jit --with-pcre2

**Comment by zhaoshikui** [comment]
My compilation parameters only added -- with pcre2

May be we can compile normally, but we cannot use it。
When I use this library, prompt me：
libpcre.so.1: cannot open shared object file: No such file or directory.

**Comment by martinhsv** [comment]
Thanks, all, for the various comments and observations.

I cannot reproduce what has been described above.

I mostly build in Ubuntu, and if pcre1 is not installed and ```--with-pcre2``` is specified, then the build runs to completion and the resulting .so file functions normally. If pcre1 **is** installed and ```--with-pcre2``` is specified then configure will search for the relevant files and include references to same in the Makefile ... but pcre1 is still not present when checking via ldd.

Nevertheless, to whatever extent the current situation is causing difficulty or annoyance for anyone, I have created PR #2975 that will omit the check for pcre1 entirely if pcre2 is specified. Feel free to apply the PR if you find it helps you.

I'm not sure that this will be merged immediately since this would break builds of the optional alp2 (I'm moderately sure that it is obsolete but ...)

**Comment by zhaoshikui** [comment]
Thank you all.
